R/toolAggregate.R

```diff
@@ -83,13 +90,26 @@
 #' toolAggregate(p, mapping, weight = p)
 #' # combined aggregation across two columns
 #' toolAggregate(p, mapping, to = "region+global")
+#'
 toolAggregate <- function(x, rel, weight = NULL, from = NULL, to = NULL, dim = 1, wdim = NULL, partrel = FALSE, # nolint
                           negative_weight = "warn", mixed_aggregation = FALSE, verbosity = 1) { # nolint
 
   if (!is.magpie(x)) stop("Input is not a MAgPIE object, x has to be a MAgPIE object!")
 
   comment <- getComment(x)
 
+  # create missing rel information from dimension names if argument "to" is set
+  if (missing(rel) && !is.null(to)) {
```

Comment:
Can't we just add a default argument for `rel`? That makes it much clearer that it is actually allowed to not pass it.

Reply:
I thought about it, but to me it looks more secure to do it this way. Alternative would be to set `rel = NULL` but I suspect that there might be cases where the provided mapping by accident is `NULL`. This cases should not trigger the automatic generation of a mapping.